sdk/keyvault/azsecrets/client.go

```diff
@@ -650,12 +539,12 @@ type ListPropertiesOfSecretVersionsResponse struct {
 	NextLink *string `json:"nextLink,omitempty" azure:"ro"`
 
 	// Secrets is the page's content.
-	Secrets []SecretItem `json:"value,omitempty" azure:"ro"`
+	Secrets []*SecretItem `json:"value,omitempty" azure:"ro"`
 }
 
 // create ListSecretsPage from generated pager
 func listSecretVersionsPageFromGenerated(i generated.KeyVaultClientGetSecretVersionsResponse) ListPropertiesOfSecretVersionsResponse {
-	var secrets []SecretItem
+	var secrets []*SecretItem
```

Comment:
Why do these need to be pointers? I feel like we go back and forth about when things are or are not - in this case it seems like they wouldn't need to be unless you're saying it can sometimes be nil?

Reply:
The short answer is "for consistency with our generated modules".

For the long answer, this was done for versioning tolerance and potential support for `x-nullable` in the future as we didn't want that to be a breaking change (it's not for other languages as they use reference types).

Reply:
It's future proofing. These elements won't be null today but if Key Vault makes them nullable, we'd have to make this breaking change at that time.